### PR TITLE
AggregationLayer: add support for 'ignoreProps' for checking aggregation

### PR DESCRIPTION
**Pull request by 1chandu**
For #3844 

#### Background
More details in #3844 

#### Change List
- LayerExtension: add needsRedraw()

### CODE REVIEW

File: modules/core/src/lib/layer-extension.js [hunk 1]
```diff
@@ -62,6 +63,21 @@ export class LayerExtension {
 
   initializeState(context, extension) {}
 
+  // Called with old and current props to determine if layer needs to redraw
+  needsRedraw(opts) {
+    const {redrawProps} = this;
```
Comment (Pessimistress) [inline modules/core/src/lib/layer-extension.js]: Can we just use the default props of the current extension? What is the use case where we need a separate list?

**Comment by 1chandu** [comment]
Based on discussion with @Pessimistress , I will update this PR with following :
Define a list of props (when changed don't require re-aggregation) and during update cycle, compare all props excluding above, to determine if re-aggregation is needed, this way we can avoid changes to the `LayerExtensions`.

File: modules/aggregation-layers/src/aggregation-layer.js [hunk 2]
```diff
@@ -18,26 +18,14 @@
...
-  'filterEnabled',
-  'filterRange',
-  'filterSoftRange',
-  'filterTransformSize',
-  'filterTransformColor'
-];
-
 export default class AggregationLayer extends CompositeLayer {
-  initializeState(aggregationProps = []) {
+  initializeState(ignoreProps = {}) {
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/aggregation-layer.js]: You are filtering `defaultProps` in each sub layer to generate this argument. There are several potential issues:
- `defaultProps` may contain `[key]: 0` or `[key]: null` which will defeat the check in `compareProps`
- we may have multiple layers of inheritance in the future, and this process only accounts for the leaf class and the base Layer

If you instead pass the aggregation props array here, you can perform the filtering with `filterProps(this.constructor._propTypes, aggregationProps)`. `constructor._propTypes` is always normalized and merged up the inheritance chain.

**Comment by 1chandu** [comment]
This PR might change, based on how my CPU Aggregation refactor work going, will wait couple more days before updating this.

File: modules/aggregation-layers/src/aggregation-layer.js [hunk 3]
```diff
@@ -87,16 +75,17 @@ export default class AggregationLayer extends CompositeLayer {
...
       return true;
     }
-    const {aggregationProps} = this.state;
-    const oldProps = {};
-    const props = {};
-    for (const propName of aggregationProps) {
-      oldProps[propName] = opts.oldProps[propName];
-      props[propName] = opts.props[propName];
+    if (
+      compareProps({
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/aggregation-layer.js]: Nit: `return compareProps(...)`

File: modules/aggregation-layers/src/utils/prop-utils.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+export function filterProps(props, filterKeys) {
+  return Object.keys(props).reduce((accu, cur) => {
+    if (!filterKeys.includes(cur)) {
+      accu[cur] = props[cur];
```
Comment (Pessimistress) [inline modules/aggregation-layers/src/utils/prop-utils.js]: If `props[cur]` is falsy the key will not be ignored in `compareProps`

**Comment by 1chandu** [comment]
Closing in favor of a better solution : https://github.com/uber/deck.gl/pull/4008, will address remaining comments in 4008.

**Pull request closed without merging**